src/pyatsimagebuilder/utils.py

```diff
@@ -219,93 +220,125 @@ def is_pyats_job(job_file):
     return False
 
 
-def to_image_path(path, context, workspace_dir):
+def search_regex(regexes, path, ignore_folders=[]):
+        regexes = [re.compile(regex) for regex in regexes]
+        ignore_folders = [path / i for i in ignore_folders]
+
+        match = []
+
+        for file in path.rglob('*'):
+            if any(i in file.parents for i in ignore_folders):
+                continue
+            if any(regex.match(file.name) for regex in regexes):
+                match.append(file)
+
+        return match
+
+
+def to_image_path(path, search_path, workspace_dir):
     '''
-    returns the relative path within image workspace
+    returns the path within image workspace
 
     Arguments:
-        path (str): Path to convert
-        context (Context): container build context
+        path (Path): Path to convert
+        search_path (Path): pathlib Path object with the directory to start discovery from
         workspace_dir (str): workspace directory
     '''
 
-    context_path = str(context.path)
+    search_path = str(search_path)
     path = str(path)
+    workspace_dir = str(workspace_dir)
 
     if path.startswith('${WORKSPACE}'):
         path = path.replace('${WORKSPACE}', workspace_dir)
     elif path.startswith('$WORKSPACE'):
         path = path.replace('$WORKSPACE', workspace_dir)
-    elif path.startswith(context_path):
-        path = path.replace(context_path, workspace_dir)
+    elif path.startswith(search_path):
+        path = path.replace(search_path, workspace_dir)
 
     return path
 
 
-def discover_jobs(jobfiles, context, install_dir, workspace_dir):
+def discover_jobs(jobfiles, 
+                  search_path, 
+                  ignore_folders, 
```

Comment:
```
ignore_folders=None,

if not ignore_folders:
    ignore_folders = []
```

I don't think `ignore_folders` should be mandatory.

Reply:
updated; made it optional